pkg/view/router.go

```diff
@@ -54,17 +90,159 @@ func New(c config.DashboardConfig, auth func(h http.Handler) http.Handler) (*chi
 	})
 
 	r.Get("/connections", func(w http.ResponseWriter, r *http.Request) {
-		err := gv.Render(w, http.StatusOK, "pages/connections/index", loadModel(r))
+		user, ok := getUser(r)
+		if !ok {
+			http.Error(w, "User not found", http.StatusInternalServerError)
+			return
+		}
+		destinations, err := storageServices.Database.GetDestinations(r.Context(), user.ID)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		m := loadModel(r)
+		m.Connections = Connections{
+			Destinations: destinations,
+		}
+		err = gv.Render(w, http.StatusOK, "pages/connections/index", m)
 		if err != nil {
 			http.Error(w, err.Error(), http.StatusInternalServerError)
 		}
 	})
 
 	r.Get("/connections/new", func(w http.ResponseWriter, r *http.Request) {
-		err := gv.Render(w, http.StatusOK, "pages/connections/new", loadModel(r))
+		err := gv.Render(w, http.StatusOK, "pages/connections/upsert", loadModel(r))
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+		}
+	})
+
+	r.Post("/connections/new", func(w http.ResponseWriter, r *http.Request) {
```

Comment:
In an ideal world, the logic here would be decoupled from this route. The next feature (next week's demo) is for users give plaid-like links to their customers to fill in their own connection details. I'd love to be able to re-use the same logic without much ado.

Reply:
The flow of development that I generally practice for things like this is breaking code shared code into functions. Dependencies declared in the parameters, and no side effects, ie. functional code.

This makes the function very portable and can be easily refactored to anywhere else in the code. As multiple of these functions seem to want to be grouped together (similar dependencies), they get their own service (type Service struct {}).

idk if this makes sense, but the goal is to optimize for velocity while also balancing abstraction needs.